src/ServerRequest.php

```diff
@@ -185,10 +185,28 @@ public static function fromGlobals()
      *
      * @return UriInterface
      */
-    public static function getUriFromGlobals() {
+    public static function getUriFromGlobals()
+    {
         $uri = new Uri('');
 
-        $uri = $uri->withScheme(!empty($_SERVER['HTTPS']) && $_SERVER['HTTPS'] !== 'off' ? 'https' : 'http');
+        $httpsKeys = [
+            'HTTP_X_FORWARDED_PROTOCOL' => 'http',
+            'HTTP_X_FORWARDED_PROTO'    => 'http',
+            'HTTP_X_FORWARDED_SSL'      => 'off',
+            'HTTP_FRONT_END_HTTPS'      => 'off',
+            'HTTP_X_URL_SCHEME'         => 'http',
+            'HTTPS'                     => 'off',
+        ];
+
+        $scheme = 'http';
+        foreach ($httpsKeys as $key => $value) {
+            if (isset($_SERVER[$key]) && $_SERVER[$key] != $value) {
```

Comment:
I wonder: is `isset` the right choice here? What if a header value is empty? In some cases it might have an obvious fallback case in some cases it might just be some error and it shouldn't be checked in the first place.

What do you think?